lib/Try/Tiny.pm

```diff
@@ -455,6 +455,27 @@ The destructor is called when the stack is unwound, after C<die> sets C<$@> to
 C<"foo at Foo.pm line 42\n">, so by the time C<if ( $@ )> is evaluated it has
 been cleared by C<eval> in the destructor.
 
+=head2 $@ might be a true value when eval succeeded
+
+Similar as in above example. It is possible to manually set C<$@> value and if
+it happens when the stack is unwound it propagate false information back to
+caller.
+
+  sub Object::DESTROY {
+    $@ = "some string";
+  }
+
+  eval {
+    my $obj = Object->new;
+  };
+
+  if ( $@ ) {
+
+  }
+
+In this case after C<eval> the value in C<$@> is set to C<"some string"> and
+evaluated to true. Even C<eval> itself succeeded.
+
 The workaround for this is even uglier than the previous ones. Even though we
```

Comment:
This paragraph was supposed to go with the previous section -- I don't think it applies to the new one?

Reply:
Why? Paragraph `The workaround for this is even uglier than the previous ones.` applies to both `=head2 $@ might not be a true value` and `=head2 $@ might be a true value when eval succeeded`.